openpype/hosts/maya/api/lib.py

```diff
@@ -3129,31 +3137,52 @@ def remove_render_layer_observer():
         pass
 
 
-def update_content_on_context_change():
-    """
-    This will update scene content to match new asset on context change
+def update_instances_frame_range():
+    """Update 'frameStart', 'frameEnd', 'handleStart', 'handleEnd' and 'asset'
+    attributes of sets that got one, execpt if instance family is 'render'
     """
-    scene_sets = cmds.listSets(allSets=True)
+    collected_instances = cmds.ls(
+        "*.id",
+        long=True,
+        type="objectSet",
+        recursive=True,
+        objectsOnly=True
+    )
     asset_doc = get_current_project_asset()
-    new_asset = asset_doc["name"]
-    new_data = asset_doc["data"]
-    for s in scene_sets:
-        try:
-            if cmds.getAttr("{}.id".format(s)) == "pyblish.avalon.instance":
-                attr = cmds.listAttr(s)
-                print(s)
-                if "asset" in attr:
-                    print("  - setting asset to: [ {} ]".format(new_asset))
-                    cmds.setAttr("{}.asset".format(s),
-                                 new_asset, type="string")
-                if "frameStart" in attr:
-                    cmds.setAttr("{}.frameStart".format(s),
-                                 new_data["frameStart"])
-                if "frameEnd" in attr:
-                    cmds.setAttr("{}.frameEnd".format(s),
-                                 new_data["frameEnd"],)
-        except ValueError:
-            pass
+
+    asset_data = asset_doc["data"]
+    frames_attributes = {
+        'frameStart': asset_data["frameStart"],
+        'frameEnd': asset_data["frameEnd"],
+        'handleStart': asset_data["handleStart"],
+        'handleEnd': asset_data["handleEnd"],
+        'asset': asset_doc['name']
+    }
+
+    for instance in collected_instances:
+        id_attr = "{}.id".format(instance)
+        if cmds.getAttr(id_attr) != "pyblish.avalon.instance":
+            continue
+
+        if (
+            cmds.attributeQuery('family', node=instance, exists=True) and
+            cmds.getAttr("{}.family".format(instance)) == "render"
+        ):
+            continue
```

Comment:
Doesn't the render instance **not** have the attributes on the instance at all either and _thus_ this check is now redundant since we're doing the attribute query check below?

Reply:
You're right